apps/admin/frontend/src/components/import_cvrfiles_modal.tsx

```diff
@@ -140,62 +132,15 @@ export function ImportCvrFilesModal({ onClose }: Props): JSX.Element | null {
     importCastVoteRecordFile(file.path);
   };
 
-  async function fetchFilenames() {
-    setCurrentState({ state: 'loading' });
-    const usbPath = await usbstick.getPath();
-    try {
-      assert(typeof usbPath !== 'undefined');
-      assert(window.kiosk);
-      const files = await window.kiosk.getFileSystemEntries(
-        join(
-          usbPath,
-          SCANNER_RESULTS_FOLDER,
-          generateElectionBasedSubfolderName(election, electionHash)
-        )
-      );
-      const newFoundFiles = files.filter(
-        (f) => f.type === 1 && f.name.endsWith('.jsonl')
-      );
-      assert(electionDefinition !== undefined);
-      const parsedFileInformation =
-        CastVoteRecordFiles.parseAllFromFileSystemEntries(newFoundFiles).sort(
-          (a, b) => b.exportTimestamp.getTime() - a.exportTimestamp.getTime()
-        );
-      setFoundFiles(parsedFileInformation);
-      await logger.log(LogEventId.CvrFilesReadFromUsb, userRole, {
-        message: `Found ${newFoundFiles.length} CVR files on USB drive, user shown option to load.`,
-        disposition: 'success',
-      });
-      setCurrentState({ state: 'init' });
-    } catch (err) {
-      assert(err instanceof Error);
-      if (err.message.includes('ENOENT')) {
-        // No files found
-        setFoundFiles([]);
-        setCurrentState({ state: 'init' });
-        await logger.log(LogEventId.CvrFilesReadFromUsb, userRole, {
-          message:
-            'No CVR files automatically found on USB drive. User allowed to load manually.',
-          disposition: 'success',
-          result: 'User allowed to manually select files.',
-        });
-      } else {
-        await logger.log(LogEventId.CvrFilesReadFromUsb, userRole, {
-          message: 'Error searching USB for CVR files.',
-          disposition: 'failure',
-          result: 'Error shown to user.',
-        });
-        throw err;
-      }
-    }
-  }
-
-  useEffect(() => {
+  // TODO: Rather than explicitly refetching, which is outside of the standard
+  // React Query pattern, we should invalidate the query on USB drive status
+  // change. Currently React Query is not managing USB drive status and polling
+  // is being performed elsewhere.
+  useExternalStateChangeListener(usbDrive.status, () => {
     if (usbDrive.status === 'mounted') {
-      void fetchFilenames();
+      void cvrFilesOnUsbQuery.refetch();
```

Comment:
Any reason you chose `query.refetch()` over `queryClient.invalidateQueries()` here? I'm not sure which would be more appropriate, but I would have defaulted to the latter

Reply:
Mostly because I feel confident that `query.refetch()` will trigger a refetch but I wasn't sure if invalidating a query while it was already mounted would trigger a refetch. But I'm realizing that we already do that in some places, so it would definitely work. I don't really have an opinion here, seems like they both work.

Reply:
Found some discussion here: https://github.com/TanStack/query/discussions/2468.

Doesn't seem like it matters!